src/Medidata.ZipkinTracer.Core/TraceProvider.cs

```diff
@@ -72,11 +73,11 @@ internal TraceProvider(IZipkinConfig config, IOwinContext context = null)
                 headerSampled = context.Request.Headers[SampledHeaderName];
             }
 
-            TraceId = Parse(headerTraceId) ? headerTraceId : GenerateHexEncodedInt64FromNewGuid();
-            SpanId = Parse(headerSpanId) ? headerSpanId : TraceId;
-            ParentSpanId = Parse(headerParentSpanId) ? headerParentSpanId : string.Empty;
+            TraceId = headerTraceId.IsParsableTo128Or64Bit() ? headerTraceId : GenerateHexFromNewGuid(config.Create128BitTraceId);
```

Comment:
I don't think we need to check the validity of the trace id here (`IsParsableTo128Or64Bit`).
The only thing that we need to check is whether `headerTraceId` is null or empty and generate a new one if it does.
Generating a new one due to invalid will break the request flow.
We should just give it to Zipkin server and if it blows up we can at least trace it back to the original request that created the invalid trace id.

What do you think @adriancole, @jcarres-mdsol, @kenyamat, @lschreck-mdsol?